scripts/gis-material-update/process/modules/cycling_infra.py

```diff
@@ -0,0 +1,76 @@
+import geopandas as gpd
+import pandas as pd
+from sqlalchemy import create_engine, text
+
+from modules.config import Config
+from modules.gis_processing import GisProcessor
+
+
+class CycleInfra(GisProcessor):
+    """Process cycle route infra."""
+
+    def __init__(self, cfg: Config):
+        self._cfg = cfg
+        self._process_result_lines = None
+        self._process_result_polygons = None
+        self._debug_result_lines = None
+
+        file_name = cfg.local_file("cycle_infra")
+
+        self._lines = gpd.read_file(file_name)
+
+    def process(self):
+        self._process_result_lines = self._lines
+
+        # Buffering configuration
+        buffers = self._cfg.buffer("cycle_infra")
+        if len(buffers) != 1:
+            raise ValueError("Unknown number of buffer values")
+
+        # buffer lines
+        target_infra_polys = self._process_result_lines.copy()
+        target_infra_polys["geometry"] = target_infra_polys.buffer(buffers[0])
+
+        # save to instance
+        self._process_result_polygons = target_infra_polys
+
+    def persist_to_database(self):
```

Comment:
I'm a bit confused. Why do we persist to DB while processing? I get similar-looking output after commenting out the to_postgis calls.

I thought the DB is used during processing in some fashion, but at least here it's not. So why is having a running PostGIS required for running the processing script?

Reply:
Good question. All processing results are saved to database after processing, during script run. Initial idea was to enable easier debugging and development (sometimes with richer attribute set than production material versions).

So far need for development in local database has been marginal.

There are at least a couple of scenarios, how to proceed:
* as is. Continue development without any change
* strip database save completely now, including already implemented data processing
* continue as is, but review the need to save materials closer to production time (anticipating a need for debugging produced materials and related algorithms)